StartEycaActivation/handler.ts

```diff
@@ -0,0 +1,270 @@
+      );
+  }
+};
+
+/**
+ * Check if a citizen is eligible for EYCA activation
+ * A citizen is eligible for EYCA while he's from 18 to 30 years old
+ * and it has already activated a CGN
+ * @param fiscalCode: the citizen's fiscalCode
+ */
+const getEycaEligibleTask = (
+  fiscalCode: FiscalCode,
+  userCgnModel: UserCgnModel
+): TaskEither<
+  IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized,
+  true
+> =>
+  fromEither(isEycaEligible(fiscalCode)).foldTaskEither<
+    IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized,
+    true
+  >(
+    () =>
+      fromLeft(ResponseErrorInternal("Cannot perform EYCA Eligibility Check")),
+    isEligible =>
+      isEligible
+        ? // If is Eligible for Eyca check if a CGN already exists (an it is ACTIVATED)
+          userCgnModel
+            .findLastVersionByModelId([fiscalCode])
+            .mapLeft<
+              IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized
+            >(() => ResponseErrorInternal("Cannot query CGN data"))
+            .chain(maybeUserCgn =>
+              maybeUserCgn
+                .foldL(
+                  () => fromLeft(ResponseErrorForbiddenNotAuthorized),
+                  userCgn =>
+                    fromPredicate(
+                      (card: Card) =>
+                        card.status === ActivatedStatusEnum.ACTIVATED,
+                      () => ResponseErrorForbiddenNotAuthorized
+                    )(userCgn.card)
+                )
+                .map(() => isEligible)
+            )
+        : fromLeft(ResponseErrorForbiddenNotAuthorized)
+  );
+
+export function StartEycaActivationHandler(
+  userEycaCardModel: UserEycaCardModel,
+  userCgnModel: UserCgnModel,
+  logPrefix: string = "StartEycaActivationHandler"
+): IStartCgnActivationHandler {
+  return async (context, fiscalCode) => {
+    const client = df.getClient(context);
+    const orchestratorId = makeEycaOrchestratorId(
+      fiscalCode,
+      PendingStatusEnum.PENDING
+    ) as NonEmptyString;
+
+    const isEycaEligibleOrError = await getEycaEligibleTask(
+      fiscalCode,
+      userCgnModel
+    ).run();
+    if (isLeft(isEycaEligibleOrError)) {
+      return isEycaEligibleOrError.value;
+    }
+
+    const card: CardPending = {
+      status: PendingStatusEnum.PENDING
+    };
+
+    return userEycaCardModel
+      .findLastVersionByModelId([fiscalCode])
+      .mapLeft<ErrorTypes | IResponseSuccessAccepted>(() =>
+        ResponseErrorInternal("Cannot query EYCA data")
+      )
+      .chain(maybeUserEycaCard =>
+        maybeUserEycaCard.foldL(
+          () => taskEither.of(fiscalCode),
+          userEycaCard =>
+            // if an EYCA card is already in a final state we return Conflict
+            [
+              ActivatedStatusEnum.ACTIVATED.toString(),
+              ExpiredStatusEnum.EXPIRED.toString(),
+              RevokedStatusEnum.REVOKED.toString()
+            ].includes(userEycaCard.card.status)
```

Comment:
Non-final state is only `Pening`?
```suggestion
           !CardPending.is(userEycaCard.card)
```